src/figuresClasses.ts

```diff
@@ -1,19 +1,73 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Can not be less than 0!');
+    }
+
+    if (a >= b + c || b >= a + c || c >= a + b) {
+      throw new Error('Side can not be bigger than sum of two others!');
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
```

Comment:
It's rather semiPerimetr
```suggestion
    const semiPerimetr = (this.a + this.b + this.c) / 2;
```